.cbmc-batch/include/proof_helpers/proof_allocators.h

```diff
@@ -17,36 +17,37 @@
 #include <aws/common/common.h>
 
 /*
- * CBMC has an internal representation in which each object has an index and an offset
+ * CBMC has an internal representation in which each object has an index and a (signed) offset
  * A buffer cannot be larger than the max size of the offset
  * The Makefile is expected to set CBMC_OBJECT_BITS to the value of --object-bits
  */
 #define MAX_MALLOC (SIZE_MAX >> (CBMC_OBJECT_BITS + 1))
 
 /**
- * The standard allocator in CBMC cannot fail.
- * This one can, which allows us to
- * nondeterministically find more bugs
+ * CBMC model of malloc always succeeds, even if the requested size is larger
+ * than CBMC can internally represent.  This function does a
+ *     __CPROVER_assume(size <= MAX_MALLOC);
+ * before calling malloc, and hence will never return an invalid pointer.
  */
-struct aws_allocator *can_fail_allocator();
+void *bounded_malloc(size_t size);
 
-static void *can_fail_malloc_allocator(struct aws_allocator *allocator, size_t size);
+/**
+ * Implemenation of aws_allocator specalized to allow CBMC to find
+ * as many bugs as possible
+ */
+struct aws_allocator *can_fail_allocator();
 
+/**
+ * CBMC model of malloc never returns NULL, which can mask bugs in C programs. Thus function:
+ * 1) Deterministically returns NULL more memory is requested than CBMC can represent
+ * 2) Nondeterminstically returns either valid memory or NULL otherwise
+ */
 void *can_fail_malloc(size_t size);
 
 /**
- * CBMC considers malloc always successed for any given size. However, a real machine
- * can only provide the available size from the pointer until the end of the address space.
- * This function models the real machine behaviour.
+ * CBMC model of realloc never returns NULL, which can mask bugs in C programs. Thus function:
+ * 1) Deterministically returns NULL more memory is requested than CBMC can represent
```

Comment:
Nit: Deterministically returns NULL `if` more memory is requested

Reply:
Done